upload.go

```diff
@@ -0,0 +1,85 @@
+package attache
+
+import (
+	"bytes"
+	"fmt"
+	"image"
+	_ "image/gif"
+	_ "image/jpeg"
+	_ "image/png"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rwcarlsen/goexif/exif"
+)
+
+func (s Server) handleUpload(w http.ResponseWriter, r *http.Request) (result uploadResponse, err error) {
+	stream := &bytes.Buffer{}
+	_, err = io.Copy(stream, r.Body)
+	if err != nil {
+		return result, errors.Wrapf(err, "copy")
+	}
+	defer r.Body.Close()
+
+	file := bytes.NewReader(stream.Bytes())
+	fileType := http.DetectContentType(stream.Bytes())
+	fileLen := stream.Len()
+
+	filePath, err := s.Storage.Upload(file, fileType)
+	if err != nil {
+		return result, errors.Wrapf(err, "upload")
+	}
+
+	fileMeta := meta(file, fileType)
+
+	result = uploadResponse{
+		Path:        filePath,
+		ContentType: fileType,
+		Bytes:       fileLen,
+		Meta:        fileMeta,
+	}
+
+	return result, nil
+}
+
+func meta(file *bytes.Reader, fileType string) uploadMeta {
+	fileMeta := uploadMeta{DateTime: "", LatLong: "", Geometry: ""}
```

Comment:
either would suffice (instead of specifying empty values for each attribute)

``` go
var fileMeta uploadMeta
// or
fileMeta :=  uploadMeta{}
```

Reply:
🆗 !